### PR TITLE
sort extensions list by name

### PR DESCRIPTION
**Pull request by JoolsMcFly**
@spywhere 's contribution to #12910 sorts the initial list of extensions by name.

This PR adds an entry to the sort by menu so users can sort by name again in case they have sorted by installs or ratings.

### CODE REVIEW

**Comment by JoolsMcFly** [comment]
I am a bit stumped as to why travis build fails. When I run VScode from my cloned repo (through `./scripts/code.sh`) I see the new "Sort by: name" menu item and no errors are thrown.

I ran `./scripts/test.sh`and got green.

Any idea?

Thanks!

ps: when I use Debug > Launch VSCode, extensions list remains empty no matter what I search. Is that expected behaviour?

**Comment by JoolsMcFly** [comment]
And now both are green. ^^

File: src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts [hunk 1]
```diff
@@ -170,6 +170,7 @@ export class ExtensionsViewlet extends Viewlet implements IExtensionsViewlet {
 				new Separator(),
 				this.instantiationService.createInstance(ChangeSortAction, 'extensions.sort.install', localize('sort by installs', "Sort By: Install Count"), this.onSearchChange, 'installs', undefined),
 				this.instantiationService.createInstance(ChangeSortAction, 'extensions.sort.rating', localize('sort by rating', "Sort By: Rating"), this.onSearchChange, 'rating', undefined),
+				this.instantiationService.createInstance(ChangeSortAction, 'extensions.sort.name', localize('sort by name', "Sort By: Name"), this.onSearchChange, 'name', undefined),
```
Comment (joaomoreno) [inline src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts]: Are you sure this alone fixes it? Did you give it a try?

**Review by joaomoreno — CHANGES_REQUESTED** [review]

**Comment by JoolsMcFly** [comment]
Hi Joao,

thanks for your answer.
I did give it a try, see attached screenshot. 
extensions

New menu item shows up and clicking on it does interact with search box. However, no matter what I search list remains empty.

I decided to try with a fresh cloned repo thinking my changes broek something. So I cloned vscode again, ran `./scripts/npm.sh install` and ran `./scripts/code.sh` to see what UI looked like.
Same thing, no extensions pop up no matter what I search.

So I went for another test. I packaged it all using `gulp vscode-linux-x64` and ran that: same thing, empty list.

If I try with an official build then I do get extensions as I search them.

Any ideas why this is?

Thanks!

**Comment by joaomoreno** [comment]
Add the following section to the `product.json` top level file:

```
	"extensionsGallery": {
		"serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",
		"cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",
		"itemUrl": "https://marketplace.visualstudio.com/items"
	}
```

This will let you run the extension gallery in development mode.

**Comment by JoolsMcFly** [comment]
Thanks a lot. I'll be afk for about two weeks and I'll give it a go when I'm back.

**Comment by JoolsMcFly** [comment]
Thanks a lot. I'll be afk for about two weeks and I'll give it a go when I'm back.






On Fri, Oct 21, 2016 at 8:57 AM +0200, "João Moreno" <notifications@github.com> wrote:





Add the following section to the `product.json` top level file:

```
        "extensionsGallery": {
                "serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",
                "cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",
                "itemUrl": "https://marketplace.visualstudio.com/items"
        }
```

This will let you run the extension gallery in development mode.

**Comment by joaomoreno** [comment]
Closing as it's been a while.

**Pull request closed without merging**